File: app/src/main/java/com/bajicdusko/fragmentmanager/javasample/SecondFragment.java
package com.bajicdusko.fragmentmanager.javasample;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.bajicdusko.fragmentmanager.IFragment;
import com.bajicdusko.fragmentmanager.R;
import com.bajicdusko.fragmentmanager.SFMFragment;

/**
 * Created by Dusko Bajic on 24.06.17.
 * GitHub @bajicdusko
 */

public class SecondFragment extends SFMFragment<ExampleFragmentChannel> implements IFragment {
    private static final String FRAGMENT_NAME = "SecondJava";

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_second, container, false);
    }

    @Override
    public void onResume() {
        super.onResume();
        setTitle();
    }

    @Override
    public String getFragmentName() {
        return FRAGMENT_NAME;
    }

    @Override
    public void setTitle() {
        fragmenChannel.setToolbarTitle(R.string.second_java);
    }

    @Override
    public void dispose() {

    }

    public static SecondFragment newInstance() {
        return new SecondFragment();
    }
}

File: app/src/main/java/com/bajicdusko/fragmentmanager/kotlinsample/SecondFragment.kt
package com.bajicdusko.fragmentmanager.kotlinsample

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.bajicdusko.fragmentmanager.R
import com.bajicdusko.kotlinfragmentmanager.SFMFragment


/**
 * Created by Dusko Bajic on 14.06.17.
 * GitHub @bajicdusko
 */
class SecondFragment : SFMFragment<ExampleFragmentChannel>() {
    private val FRAGMENT_NAME: String = "Second"

  override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
      savedInstanceState: Bundle?): View? = inflater.inflate(R.layout.fragment_second, container,
      false)

    override fun onResume() {
        super.onResume()
        setTitle()
    }

    override fun getFragmentName(): String = FRAGMENT_NAME

    override fun setTitle(): Unit? = fragmentChannel?.setToolbarTitle(R.string.second)

    override fun dispose() {
        //Clear the resources
    }

    companion object {
        fun newInstance(): SecondFragment = SecondFragment()
    }
}